app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentPresenterImpl.kt

```diff
@@ -264,4 +264,16 @@ class EnrollmentPresenterImpl(
 
     fun getEventStage(eventUid: String) =
         enrollmentFormRepository.getProgramStageUidFromEvent(eventUid)
+
+    fun showOrHideSaveButton() {
+        val teiUid = teiRepository.blockingGet().uid()
+        val programUid = getProgram().uid()
+        val hasEnrollmentAccess = d2.enrollmentModule().enrollmentService()
```

Comment:
I would extract D2 usage from our business logic in order to improve clarity and facilitate testing, why do  you think about extracting this to a class like IsSaveButtonVisible or CheckEnrollmentAcces